yarn-project/simulator/src/public/executor.ts

```diff
@@ -43,7 +44,10 @@ export class PublicExecutor {
     const address = execution.contractAddress;
     const selector = execution.functionSelector;
     const startGas = availableGas;
-    const fnName = (await this.contractsDb.getDebugFunctionName(address, selector)) ?? `${address}:${selector}`;
+    // this will be passed to the prover
+    let fnName = (await this.contractsDb.getDebugFunctionName(address, selector)) ?? `${address}:${selector}`;
+    // add an identifier to the function name
```

Comment:
If you merge this one (the change in function name) since it will mess up the benchmarks. This name is used originally used for aggregation of simulation times, and for proving time. If we add a unique id, then it will not correctly group the calls to the same function across the e2e tests (and across many e2e tests).

Maybe it's not a big deal, but probably the right way to do it is to define some kind of separate property "call_id/execution_id/trace_id" or similar, use it and pass it all around. It might be a bit a bit complicated (a bit more complex than this [PR](https://github.com/AztecProtocol/aztec-packages/pull/7084)) so I can do it if you'd want.

You can still keep the logging of nested calls.

Reply:
I will remove this. I think that the function IDs. in the nested calls are sufficient to debug.